src/imars3d/backend/workflow/validate.py

```diff
@@ -0,0 +1,122 @@
+import importlib
+from importlib.util import find_spec
+import json
+import jsonschema
+from pathlib import Path
+import sys
+from typing import Any, Dict, Tuple, Union
+
+FilePath = Union[Path, str]
+
+# http://json-schema.org/learn/getting-started-step-by-step.html
+SCHEMA_FILE = Path(__file__).parent / "schema.json"
+
+
+def _load_schema():
+    try:
+        with open(SCHEMA_FILE, "r") as handle:
+            schema = json.load(handle)
+    except FileNotFoundError as e:
+        raise RuntimeError(f"Failed to load schema from {SCHEMA_FILE}") from e
+
+    if not schema:
+        raise RuntimeError(f"Failed to load schema from {SCHEMA_FILE}")
+    return schema
+
+
+SCHEMA = _load_schema()
```

Comment:
I suppose this is one way to create a singleton from file.

Reply:
It's weird that `_load_schema()` can only be used once, one of Pete's incantations